src/libAtomVM/opcodesswitch.h

```diff
@@ -6660,8 +6663,11 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
                             DECODE_LITERAL(stride, code, i, next_off);
                             j++;
                             #ifdef IMPL_EXECUTE_LOOP
+                                if (stride < 0) {
+                                    RAISE_ERROR(BADARG_ATOM);
+                                }
                                 size_t bs_bin_size = term_binary_size(bs_bin);
-                                if ((bs_bin_size * 8) - bs_offset != stride) {
+                                if ((bs_bin_size * 8) - bs_offset != (size_t) stride) {
```

Comment:
here the same, let's try avoid casts.

Reply:
A cast is required here, but only after a check to make sure the value is not < 0.  But the code has been updated to be more clear.